### ISSUE TITLE
Open folders of irregularly shaped images

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#605]
## 🚀 Feature
It would be nice to be able to open a folder of irregularly shaped images using some form of lazy loading / dask.

**Comment by AhmetCanSolak** [comment]
I agree it would be nice to have, @sofroniewn  do you have a default convention to overlay images?

**Comment by sofroniewn** [comment]
I'd probably align (0,0), the top left corner by default, but you could align to the center. The thing that makes this annoying is I think its only really possible with dask array if you know all the shapes ahead of time, so you need to loop over the data once to get all the shapes, but you want to do that without loading all the image data, so you have to look at the image metadata, but then that gets annoying. 

[dask_image.imread](http://image.dask.org/en/latest/_modules/dask_image/imread.html#imread) uses [PIMS](http://soft-matter.github.io/pims/v0.3.3/opening_files.html) which I think can read metadata without reading the bytes, but it's still maybe not ideal
